### PR TITLE
Enable support for root NS records

### PR DESCRIPTION
**Pull request by rtokarek-fastly**
Enable `SUPPORTS_ROOT_NS` for the google provider. 

Google Cloud DNS allows for management of root NS records. Enabling this flag (in testing on a hand-edited octodns-googlecloud module) successfully maintained and modified the root NS record for a test domain. 

Note that the Google APIs require that there be some value (you cannot delete root NS records), but octodns-sync since version 0.9.17 has no trouble managing these records in Google Cloud DNS.

### CODE REVIEW

**Comment by rtokarek-fastly** [comment]
It wasn't clear to me whether I should be incrementing versions by hand. I'm happy to close this and resubmit if ci would mange versioning instead. Please let me know.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
-## v0.0.1 - 2022-01-11 - Moving
+## v0.0.2 - 2022-10-27
+
+* Enable support for root level NS records (`SUPPORTS_ROOT_NS=true`)
+
+## v0.0.1 - 2022-01-11
```
Comment (ross) [inline CHANGELOG.md:5]: Looks like the "title" of this release was removed as part of adding the stuff up above.
```suggestion
## v0.0.1 - 2022-01-11 - Moving
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,8 @@
-## v0.0.1 - 2022-01-11 - Moving
+## v0.0.2 - 2022-10-27
```
Comment (ross) [inline CHANGELOG.md:1]: octoDNS avoids doing releases in PRs with other changes. In this PR things can stay ??-??

```suggestion
## v0.0.2 - 2022-??-??
```

File: README.md [hunk 3]
```diff
@@ -18,8 +18,8 @@ Pinning specific versions or SHAs is recommended to avoid unplanned upgrades.
 
 ```
 # Start with the latest versions and don't just copy what's here
-octodns==0.9.14
-octodns-googlecloud==0.0.1
+octodns==0.9.21
+octodns-googlecloud==0.0.2
```
Comment (ross) [inline README.md:22]: Doesn't hurt anything to update these, but they're generally just left alone during new releases and it's assumed/hoped that people will pick good versions. Perhaps I should go through and change them all to placeholders to make that more clear. Don't need to make that change as part of this PR. I prefer 1-change PRs.

File: octodns_googlecloud/__init__.py [hunk 4]
```diff
@@ -13,7 +13,7 @@
 from octodns.record import Record
 from octodns.provider.base import BaseProvider
 
-__VERSION__ = '0.0.1'
+__VERSION__ = '0.0.2'
```
Comment (ross) [inline octodns_googlecloud/__init__.py:16]:
```suggestion
__VERSION__ = '0.0.1'
```

**Review by ross — COMMENTED** [review]
Thanks for testing/verifying that google can manage them. Few comments/suggestions inline.  Once those are sorted we can 🚢

**Comment by rtokarek-fastly** [comment]
I've reverted the things you suggested, thanks for the info! I wasn't sure how much to include in the PR or what the release process looked like.

**Comment by ross** [comment]
setup-py failure is unrelated to this PR. looks like the google protobuf & cloud libs have gotten into yet another versions are incompatible with things state.

**Comment by ross** [comment]
> setup-py failure is unrelated to this PR. looks like the google protobuf & cloud libs have gotten into yet another versions are incompatible with things state. Will dig into that this afternoon and hopefully get things unblocked.

As of yet I've had no luck figuring out what's going on here. I cannot reproduce this locally on OSX. Next step will be poking around in docker. :-/

**Comment by ross** [comment]
Was able to reproduce the problem with a python:3.10.8 docker container and protobuf version 4.21.9. Didn't seem to happen with 4.21.8 so clamped the version in https://github.com/octodns/octodns-googlecloud/pull/19.

**Pull request merged**